src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -836,6 +836,33 @@ public static ErlangFile resolveToFile(@Nullable ErlangModuleRef ref) {
     return ObjectUtils.tryCast(resolved != null ? resolved.getContainingFile() : null, ErlangFile.class);
   }
 
+  @NotNull
+  public static List<ErlangFunction> resolveToFunctions(@Nullable ErlangFunctionCallExpression call) {
+    PsiReference reference = call != null ? call.getReference() : null;
+    ResolveResult[] results = reference instanceof PsiPolyVariantReference ?
+                              ((PsiPolyVariantReference) reference).multiResolve(true) : ResolveResult.EMPTY_ARRAY;
+    return ContainerUtil.mapNotNull(results, new Function<ResolveResult, ErlangFunction>() {
+      @Override
+      @Nullable
+      public ErlangFunction fun(ResolveResult resolveResult) {
+        return ObjectUtils.tryCast(resolveResult.getElement(), ErlangFunction.class);
+      }
+    });
+  }
+
+  @NotNull
+  private static Pair<List<ErlangFunction>, Integer> resolveToFunctions(@NotNull PsiElement callArgument) {
+    ErlangExpression arg = PsiTreeUtil.getParentOfType(callArgument, ErlangExpression.class, false);
+    while (arg != null && !(arg.getParent() instanceof ErlangArgumentList)) {
+      arg = ObjectUtils.tryCast(arg.getParent(), ErlangExpression.class);
+    }
+    ErlangArgumentList argList = ObjectUtils.tryCast(arg != null ? arg.getParent() : null, ErlangArgumentList.class);
+    int position = argList != null ? argList.getExpressionList().indexOf(arg) : -1;
+    ErlangFunctionCallExpression call = ObjectUtils.tryCast(argList != null ? argList.getParent() : null,
+                                                            ErlangFunctionCallExpression.class);
+    return Pair.create(resolveToFunctions(call), position);
```

Comment:
I believe, you can rewrite your code with simpler methods, also making it way simpler to read.